PocketKit/Sources/Sync/Slates/SlateService.swift

```diff
@@ -2,31 +2,59 @@ import Apollo
 import Foundation
 
 protocol SlateService {
-    func fetchSlateLineup(_ identifier: String) async throws -> SlateLineup?
-    func fetchSlate(_ slateID: String) async throws -> Slate?
+    func fetchSlateLineup(_ identifier: String) async throws
+    func fetchSlate(_ slateID: String) async throws
 }
 
 class APISlateService: SlateService {
     private let apollo: ApolloClientProtocol
+    private let space: Space
 
-    init(apollo: ApolloClientProtocol) {
+    init(
+        apollo: ApolloClientProtocol,
+        space: Space
+    ) {
         self.apollo = apollo
+        self.space = space
     }
 
-    func fetchSlateLineup(_ identifier: String) async throws -> SlateLineup? {
+    func fetchSlateLineup(_ identifier: String) async throws {
         let query = GetSlateLineupQuery(lineupID: identifier, maxRecommendations: 5)
-        
-        return try await apollo
-            .fetch(query: query)
-            .data?.getSlateLineup
-            .flatMap { $0 }
-            .map(SlateLineup.init)
+
+        guard let remote = try await apollo.fetch(query: query).data?.getSlateLineup else {
+            return
+        }
+
+        if let existingSlateLineup = try space.fetchSlateLineup(byRemoteID: remote.id) {
+            space.delete(existingSlateLineup)
+        }
+
+        let unsavedItems = try space.fetchUnsavedItems()
+        unsavedItems.forEach {
+            space.delete($0)
+        }
+
+        let slateLineup: SlateLineup = space.new()
+        slateLineup.update(from: remote, in: space)
+
+        try self.space.save()
     }
 
-    func fetchSlate(_ slateID: String) async throws -> Slate? {
+    func fetchSlate(_ slateID: String) async throws {
         let query = GetSlateQuery(slateID: slateID, recommendationCount: 25)
-        let remote = try await apollo.fetch(query: query).data?.getSlate
 
-        return (remote?.fragments.slateParts).flatMap(Slate.init)
+        guard let remote = try await apollo.fetch(query: query).data?.getSlate?.fragments.slateParts else {
+            return
+        }
+
+        let unsavedItems = try space.fetchUnsavedItems()
```

Comment:
Let's use core data's bulk delete feature to delete all the records that match a given fetch request (see `Source.clear()` for an example)